### DISCUSSION TITLE
Integrate CrowdStrike falcon installation and OIDC setup

### DISCUSSION THREAD

**Discussion by ashirwadnivalkar** [npalm/terraform-aws-gitlab-runner#1017] [Q&A]
Hey folks 

I am new to Terraform and seeking some help/direction about how I could integrate installation of Crowdstrike falcon during the EC2 creation of this runner. Also, wanted to integrate OIDC for this runner. 
Any guidance or pointers to any documentation or examples would be helpful. 

Thank you

**Comment by kayman-mk** [comment]
Sorry, but I do not know Crowdstrike. I think they should have some docs available showing how to use it. For OIDC and Github check https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-cloud-providers

**Reply by ashirwadnivalkar** [reply]
Thanks for the document pointers for OIDC. Okay ,if not Crowdstrike, can you please guide me how I can install any package using a command on to the main Runner Orchestrator EC2 instance. for eg: passing commands via bash script which creation of EC2 instance. 
I know we can do this using `user_data` but I am unaware where I can modify this code.

**Reply by kayman-mk** [reply]
You can use the `runner_install.post_install_script` to inject a script:

```hcl
module "gitlab_runner" {
  runner_install = {
    post_install_script = templatefile("${path.module}/templates/runners_user_install_script",
      {
        gitlab_runner_version = local.gitlab_runner_version
    })
  }
}
```

Basic `runners_user_install_script`:
```
echo "post install script on the Runner"
```
